### PR TITLE
change of docker backend

### PR DESCRIPTION
**Pull request by cooktheryan**
#### What does this PR do?
Modify docker backend for overlay2

#### How should this be manually tested?
Perform an installation

#### Is there a relevant Issue open for this?
https://trello.com/c/DPwnHNy8/971-5spike-azure-ocp-use-overlay2

#### Who would you like to review this?
cc: @glennswest @e-minguez

### CODE REVIEW

**Comment by cooktheryan** [comment]
@e-minguez or @glennswest can someone do a test deploy with this?

**Comment by e-minguez** [comment]
I'll do, let me figure it out how Azure worked :D

**Comment by glennswest** [comment]
Just saw this will test tomm

**Comment by e-minguez** [comment]
I'm still not able to try your PR as I cannot deploy the ref. arch. due to quota limits and the all-in-one template doesn't work with ansible, so I need to test it manually... I've created a couple of cards to track it (update ansible code for all-in-one support) and some other issues I've found (update the ref. arch. doc for the new folder structure) and I've just realized that to test it I need to change the github template [here](https://github.com/openshift/openshift-ansible-contrib/blob/master/reference-architecture/azure-ansible/3.6/azuredeploy.json#L314) so I need to clone your changes and add that one 💃

File: reference-architecture/azure-ansible/3.6/allinone.sh [hunk 1]
```diff
@@ -83,9 +83,11 @@ DOCKERVG=$( parted -m /dev/sda print all 2>/dev/null | grep unknown | grep /dev/
 echo "DEVS=${DOCKERVG}" >> /etc/sysconfig/docker-storage-setup
 cat <<EOF > /etc/sysconfig/docker-storage-setup
 DEVS=$DOCKERVG
-VG=docker-vg
+VG=docker_vol
```
Comment (e-minguez) [inline reference-architecture/azure-ansible/3.6/allinone.sh]: That's not required, but anyway is ok.

**Pull request merged**